pact/coin-contract/coin.pact

```diff
@@ -451,6 +455,93 @@
           (credit create-account create-account-guard quantity))
         ))
     )
+
+
+  ; --------------------------------------------------------------------------
+  ; Coin allocations
+
+  (defschema allocation-schema
+    @doc "Genesis allocation registry"
+    ;@model [ (invariant (>= balance 0.0)) ]
+
+    balance:decimal
+    date:time
+    guard:guard
+    redeemed:bool)
+
+  (deftable allocation-table:{allocation-schema})
+
+  (defun create-allocation-account
+    ( account:string
+      date:time
+      guard-ref:string
+      amount:decimal
+    )
+
+    @doc "Add an entry to the coin allocation table. Requires GENESIS \
+         \capability. "
+
+    @model [ (property (valid-account account)) ]
+
+    (require-capability (GENESIS))
+
+    (validate-account account)
+    (enforce (>= amount 0.0)
+      "allocation amount must be non-negative")
+
+    (enforce-unit amount)
+
+    (let
+      ((guard:guard (keyset-ref-guard guard-ref)))
+
+      (create-account account guard)
+
+      (insert allocation-table account
+        { "balance" : amount
+        , "date" : date
+        , "guard" : (keyset-ref-guard guard-ref)
+        , "redeemed" : false
+        })))
+
+  (defun release-allocation
+    ( account:string )
+
+    @doc "Release funds associated with an allocation account"
+    @model [ (property (valid-account account)) ]
+
+    (validate-account account)
+
+    (with-read allocation-table account
+      { "balance" := balance
+      , "date" := release-time
+      , "redeemed" := redeemed
+      , "guard" := guard
+      }
+
+      (let ((curr-time:time (at 'block-time (chain-data))))
+
+        (enforce (not redeemed)
+          "allocation funds have already been redeemed")
+
+        (enforce
+          (>= (diff-time curr-time release-time) 0.0)
```

Comment:
You can simply test `(>= curr-time release-time)`

Reply:
Convenient!